inspire_disambiguation/api.py

```diff
@@ -21,127 +21,162 @@
-    clusterer.load_data(
-        current_app.config['DISAMBIGUATION_CURATED_SIGNATURES_PATH'],
-        current_app.config['DISAMBIGUATION_PUBLICATIONS_PATH'],
-        current_app.config['DISAMBIGUATION_INPUT_CLUSTERS_PATH'],
-    )
-    clusterer.fit(n_jobs=current_app.config['DISAMBIGUATION_CLUSTERING_N_JOBS'])
-    clusterer.save_model(current_app.config['DISAMBIGUATION_CLUSTERING_MODEL_PATH'])
+    clusterer.load_data(signatures, input_clusters)
+    LOGGER.info("Starting clustering")
+    clusterer.fit(n_jobs=n_jobs)
+
+    return process_clustering_output(clusterer)
+
 
+def process_clustering_output(clusterer):
+    """Process output of `Clusterer.fit` function to meet requirements of inspire.
+    Args:
+        clusterer (Clusterer): Clusterer object with all data processed by fit function.
+
+    Returns:
+        list: list with dicts
+        Examples:
+            [
+              {(publication_id, signature_uuid): [(author_id, bool to
+                  `author_has_claims`), ...]},
+              ...,
+            ]
+
+    """
+    labels = clusterer.clusterer.labels_
+    output = {}
+    for label in numpy.unique(labels):
+        signatures = clusterer.X[labels == label]
+        author_ids_and_is_curated = {}
+        for sig in signatures:
+            author_id = sig[0]["author_id"]
+            if sig[0]["is_curated_author_id"]:
+                author_ids_and_is_curated[author_id] = True
+            elif author_id and author_id not in author_ids_and_is_curated:
+                author_ids_and_is_curated[author_id] = False
+        for sig in signatures:
+            output[(sig[0].publication["publication_id"], sig[0]["signature_uuid"])] = [
+                (author_id, is_curated)
+                for author_id, is_curated in author_ids_and_is_curated.items()
+            ]
+    return output
+
+
+def cluster_from_redis(ethnicity_model_path, distance_model_path, n_jobs):
+    """
+    Process all signature blocks from redis set (one by one).
+    Args:
+        ethnicity_model_path (str): Full path where ethnicity model is saved.
+        distance_model_path (str): Full path where distance model is saved.
+        n_jobs (int): How many jobs will be running to fit data.
+
+    """
+    redis_url = conf["REDIS_URL"]
+    redis = StrictRedis.from_url(redis_url, decode_responses=True)
+    while True:
+        signature_block_data = redis.bzpopmin(
+            conf["REDIS_PHONETIC_BLOCK_KEY"], conf["REDIS_TIMEOUT"]
+        )
+        if not signature_block_data:
+            LOGGER.warning("No signature blocks in redis to process! STOP.")
+            break
+        signature_block = signature_block_data[1]
+        LOGGER.info("Processing '%s' signature_block", signature_block)
+        LOGGER.info(
+            "%s",
+            pprint.pformat(
+                cluster(
+                    ethnicity_model_path, distance_model_path, n_jobs, signature_block
+                ),
+                indent=4,
+            ),
```

Comment:
multiline logs don't look good on ES at the moment, fyi

Reply:
This is only temporary until disambiguation part2 will be finished. I'm not sure, but I believe that until then all runs of this script will be done manually so readable output is quite important.